Dockerfile

```diff
@@ -1,26 +1,16 @@
-FROM alpine:3.5
-LABEL maintainer "Mario Freitas <imkira@gmail.com>"
+FROM golang:1.12.0-alpine3.9 AS builder
+RUN apk add --no-cache git
+WORKDIR /build
+COPY . .
+RUN go build .
 
-WORKDIR /usr/local/bin
+FROM alpine:3.9
+RUN addgroup -S gcp-iap-auth \
+    && adduser -S gcp-iap-auth -G gcp-iap-auth \
+    && mkdir /app \
+    && chown gcp-iap-auth:gcp-iap-auth /app
+USER gcp-iap-auth
+WORKDIR /app
+COPY --from=builder --chown=gcp-iap-auth /build/gcp-iap-auth .
 
-ENV RELEASE_VER  v0.0.3
-ENV RELEASE_SHA1 69873d40040b5f65698d9cf59d713d0f141ca1ea
-
-RUN apk add --no-cache --update \
-      ca-certificates \
-      curl \
-    && curl -Lsj -o gcp-iap-auth https://github.com/imkira/gcp-iap-auth/releases/download/${RELEASE_VER}/gcp-iap-auth-linux-amd64 \
-    && (echo "${RELEASE_SHA1} *gcp-iap-auth" | sha1sum -c -) \
-    && chmod +x gcp-iap-auth \
-    && apk del curl \
-    && rm -rf /var/cache/apk/*
-
-#ENV GCP_IAP_AUTH_LISTEN_ADDR=0.0.0.0
-#ENV GCP_IAP_AUTH_LISTEN_PORT=80
-#ENV GCP_IAP_AUTH_AUDIENCES=https://domain1,https://domain2
-#ENV GCP_IAP_AUTH_PUBLIC_KEYS=/path/to/public_keys_file
-#ENV GCP_IAP_AUTH_TLS_CERT=/path/to/tls_cert_file
-#ENV GCP_IAP_AUTH_TLS_KEY=/path/to/tls_key_file
-
-EXPOSE 80 443
-ENTRYPOINT ["/usr/local/bin/gcp-iap-auth"]
+CMD ["/app/gcp-iap-auth"]
```

Comment:
Why cmd and not entrypoint? With entrypoint args get passed to the gcp-iap-auth program directly, without it you'd have to invoke `docker run someimage /app/gcp-iap-auth my arguments`, instead of just `docker run someimage my arguments`

I guess it doesn't really matter much, just curious if there was a reason for the change.

Reply:
The `CMD` vs `ENTRYPOINT` distinction doesn't really matter in k8s, but I generally prefer `CMD` when I'm setting a "default" command for the image because it's easier to override in the docker CLI.